src/components/clusterprovider/minikube/minikubeclusterprovider.ts

```diff
@@ -0,0 +1,224 @@
+        await this.handleCreate(request, { clusterType: request.query["clusterType"] }, response, next);
+    }
+
+    async handlePostCreate(request: restify.Request, response: restify.Response, next: restify.Next) {
+        await this.handleCreate(request, request.body, response, next);
+    }
+
+    async handleGetConfigure(request: restify.Request, response: restify.Response, next: restify.Next) {
+        await this.handleConfigure(request, { clusterType: request.query["clusterType"] }, response, next);
+    }
+
+    async handlePostConfigure(request: restify.Request, response: restify.Response, next: restify.Next) {
+        await this.handleConfigure(request, request.body, response, next);
+    }
+
+    async handleCreate(request: restify.Request, requestData: any, response: restify.Response, next: restify.Next): Promise<void> {
+        await this.handleRequest(getHandleCreateHtml, request, requestData, response, next);
+    }
+
+    async handleConfigure(request: restify.Request, requestData: any, response: restify.Response, next: restify.Next): Promise<void> {
+        await this.handleRequest(getHandleConfigureHtml, request, requestData, response, next);
+    }
+
+    async handleRequest(handler: HtmlRequestHandler, request: restify.Request, requestData: any, response: restify.Response, next: restify.Next) {
+        const html = await handler(request.query["step"], requestData, this.context);
+        response.contentType = 'text/html';
+        response.send("<html><body><style id='styleholder'></style>" + html + "</body></html>");
+
+        next();
+    }
+}
+
+async function getHandleCreateHtml(step: string | undefined, requestData: any, context: Context): Promise<string> {
+    if (!step) {
+        return await promptForConfiguration(requestData, context, "create", "create");
+    } else if (step === "create") {
+        return await createCluster(requestData, context);
+    } else if (step === "wait") {
+        return await waitForClusterAndReportConfigResult(context, requestData);
+    } else {
+        return renderInternalError(`MinikubeStepError (${step})`);
+    }
+}
+
+async function getHandleConfigureHtml(step: string | undefined, requestData: any): Promise<string> {
+    if (!step || step === "configure") {
+        return await configureKubernetes(requestData);
+    } else {
+        return renderInternalError(`ConfigurationStepError (${step})`);
+    }
+}
+
+async function promptForConfiguration(previousData: any, context: Context, action: clusterproviderregistry.ClusterProviderAction, nextStep: string): Promise<string> {
+    return formPage({
+        stepId: 'PromptForConfiguration',
+        title: 'Configure Minikube',
+        waitText: 'Configuring Minikube',
+        action: action,
+        nextStep: nextStep,
+        submitText: 'Create',
+        previousData: previousData,
+        formContent: `
+        <p>No configuration options (yet)</p>
```

Comment:
Let's provide a way to signal to the form helper that there's no content.  Or the content could be "This adds your local Minikube cluster to your Kubernetes configuration." i.e. explaining what the action link is going to do before the user clicks it (otherwise there's not much point having this page...!).